### PR TITLE
Show definition for non-lemma wordforms | Entry card UI revamp

### PR DESCRIPTION
**Pull request by Madoshakalaka**
# Super Dope

Implemented the user interface mentioned in #201 
Screenshot from 2020-01-23 16-40-00

# Changes
* definition for non-lemmas also gets shown (if they exist)
* unskipped unit tests for new search code

# Need Help
* I think I destroyed CSS. I need someone with higher aesthetics taste and CSS kung-fu to fix it @eddieantonio

### CODE REVIEW

**Comment by Madoshakalaka** [comment]
I CAN'T BELIEVE this beautiful pr failed in CI. LEMME FIX IT

File: CreeDictionary/API/models.py [hunk 1]
```diff
@@ -29,22 +30,25 @@ def filter_cw_wordforms(q: QuerySet) -> Iterable["Wordform"]:
                 break
 
 
-@attrs(auto_attribs=True)
+@attrs(auto_attribs=True, frozen=True)  # frozen makes it hashable
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: noice!

File: CreeDictionary/API/models.py [hunk 2]
```diff
@@ -29,22 +30,25 @@ def filter_cw_wordforms(q: QuerySet) -> Iterable["Wordform"]:
...
 class SearchResult:
     """
     Contains all of the information needed to display a search result.
+
+    Comment:
+    Each instance corresponds visually to one card shown on the interface
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: :+1: yeah, this is a lot nicer than before

File: CreeDictionary/API/models.py [hunk 3]
```diff
@@ -29,22 +30,25 @@ def filter_cw_wordforms(q: QuerySet) -> Iterable["Wordform"]:
...
-    part_of_speech: SimpleLC
+    matched_cree: str
 
-    # Is this the lemma?
     is_lemma: bool
 
-    # The text of the matched lemma
-    lemma: str
+    # English or Cree
+    matched_by: Language
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: YESSSSSSS!

File: CreeDictionary/API/models.py [hunk 4]
```diff
@@ -389,89 +369,91 @@ def fetch_lemma_by_user_query(user_query: str) -> "CreeAndEnglish":
...
         return CreeAndEnglish(cree_results, english_results)
 
-    @staticmethod
-    def search(user_query: str) -> Tuple[str, List[SearchResult]]:
-        """
-        Returns the matched language (as an ISO 639 code), and list of search
-        results.
-        """
-        cree_results, english_results = Wordform.fetch_lemma_by_user_query(user_query)
+    def __lt__(self, other):
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: Might consider using [`functools.total_ordering`](https://docs.python.org/3.6/library/functools.html#functools.total_ordering), but I'm not sure how that works inside a Django `Model`.

File: CreeDictionary/API/models.py [hunk 5]
```diff
@@ -389,89 +369,91 @@ def fetch_lemma_by_user_query(user_query: str) -> "CreeAndEnglish":
...
 Lemma = NewType("Lemma", Wordform)
 
 
 class CreeResult(NamedTuple):
     """
     - analysis: a string, one fst analysis of user query
 
-    - normatized_cree: a string, the Cree inflection that matches the analysis
+    - normatized_cree: a wordform, the Cree inflection that matches the analysis
+        Can be a string that's not saved in the database since our database do not store all the
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: 👍

File: CreeDictionary/API/models.py [hunk 6]
```diff
@@ -486,10 +468,38 @@ class EnglishResult(NamedTuple):
...
+    determine how we sort search results.
+
+    :return:   0: does not matter;
+              >0: res_a should appear before res_b;
+              <0: res_a should appear after res_b.
+    """
+
+    if res_a.matched_by is Language.CREE and res_b.matched_by is Language.CREE:
+        # both from cree
+        a_dis = get_modified_distance(user_query, res_a.matched_cree)
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: 👍

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/word-entries.html [hunk 7]
```diff
@@ -1,2 +1,83 @@
-{% include 'CreeDictionary/cree-entries.html' %}
-{% include 'CreeDictionary/english-entries.html' %}
\ No newline at end of file
+{% load inflection_extras %}
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/word-entries.html]: 👍 lol I'm not gonna miss `cree-entries.html` and `english-entries.html`

File: CreeDictionary/CreeDictionary/views.py [hunk 8]
```diff
@@ -30,12 +30,8 @@ def search_results(request, query_string: str):
     returns rendered boxes of search results according to user query
     """
     # todo: use the analysis from cree_results to show user query analysis (Preverb, reduplication, IC ...)
-    cree_results, english_results = Wordform.fetch_lemma_by_user_query(query_string)
-    return render(
-        request,
-        "CreeDictionary/word-entries.html",
-        {"cree_results": cree_results, "english_results": english_results},
-    )
+    results = Wordform.search(query_string)
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/views.py]: so much nicer! 🤩🤩🤩

**Comment by eddieantonio** [comment]
@kobexamoh: after this is merged, I'd like you to touch up the HTML/CSS to make this look goewgiouss!

**Comment by kobexamoh** [comment]
> @kobexamoh: after this is merged, I'd like you to touch up the HTML/CSS to make this look goewgiouss!

Noted and noted – will do 🙏

File: CreeDictionary/API/models.py (hunk 4 above)
Comment (Madoshakalaka) [reply CreeDictionary/API/models.py]: ooOOo I think this `__lt__` is old code I added and it's not needed any more. I'll clean it!

**Comment by Madoshakalaka** [comment]
yeEeeEeeeeEEEE

**Pull request merged**